src/domains/fdr-common/03_cosmos_mongodb_fdr.tf

```diff
@@ -63,105 +63,42 @@ resource "azurerm_cosmosdb_mongo_database" "fdr" {
 locals {
   collections = [
     {
-      name = "fdr_history"
-      indexes = [{
-        keys   = ["_id"] # reporting_flow_name + revision
-        unique = true
-        }
-      ]
-      shard_key = null
-    },
-    {
-      name = "fdr_insert"
+      name = "fdr_flow"
       indexes = [
         {
-          keys   = ["_id"] # reporting_flow_name
+          keys   = ["_id"] # document UID
           unique = true
         },
         {
-          keys   = ["fdr"]
-          unique = false
-        },
-        {
-          keys   = ["sender.psp_id"]
-          unique = false
-        },
-        {
-          keys   = ["fdr", "revision"]
+          keys   = ["sender.psp_id", "name", "revision"] # flow_revision_idx
           unique = true
-        }
-      ]
-      shard_key = null
-    },
-    {
-      name = "fdr_publish"
-      indexes = [{
-        keys   = ["_id"] # reporting_flow_name
-        unique = true
-        },
-        {
-          keys   = ["fdr"]
-          unique = false
         },
         {
-          keys   = ["sender.psp_id"]
+          keys   = ["receiver.organization_id", "sender.psp_id", "published"] # published_flow_idx
           unique = false
-        },
-        {
-          keys   = ["receiver.organization_id"]
-          unique = false
-        },
-        {
-          keys   = ["fdr", "revision"]
-          unique = true
-        }
-      ]
-      shard_key = null
-    },
-    {
-      name = "fdr_payment_history"
-      indexes = [{
-        keys   = ["_id"] # index + ref_fdr_reporting_flow_name + ref_fdr_id
-        unique = true
         }
       ]
-      shard_key = null
+      shard_key   = null,
+      ttl_seconds = "3024000" # 30 days + 5 days (deltaTime)
```

Comment:
Could be useful export it as variable?
e.g. for different environment can be useful have different TTL

Reply:
Good idea, I will set TTL values as environment variables in the next commit.